File: app/src/main/java/kyklab/burner2/utils/FMUtils.java
package kyklab.burner2.utils;

import android.content.Context;
import android.net.Uri;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

public class FMUtils {
    private static final String TAG = "FMUtils";

    public static void copy(InputStream inputStream, OutputStream outputStream)
            throws IOException {
        byte[] buf = new byte[1024];
        int len;
        while ((len = inputStream.read(buf)) > 0) {
            outputStream.write(buf, 0, len);
        }
    }

    public static void copy(String srcPath, String destPath) throws IOException {
        try (InputStream is = new FileInputStream(srcPath);
             OutputStream os = new FileOutputStream(destPath)) {
            copy(is, os);
        } catch (IOException e) {
            e.printStackTrace();
            throw e;
        }
    }

    public static void copy(Context context, Uri srcUri, String destPath) throws IOException {
        try (InputStream is = context.getContentResolver().openInputStream(srcUri);
             OutputStream os = new FileOutputStream(destPath)) {
            if (is != null) {
                copy(is, os);
            } else {
                throw new IOException("InputStream is null");
            }
        } catch (IOException e) {
            e.printStackTrace();
            throw e;
        }
    }
}

File: app/src/main/java/kyklab/burner2/utils/FMUtils.kt
package kyklab.burner2.utils

import android.content.Context
import android.net.Uri
import java.io.*

object FMUtils {
    private val TAG = "FMUtils"

    @Throws(IOException::class)
    fun copy(inputStream: InputStream, outputStream: OutputStream) {
        val buf = ByteArray(1024)
        var len: Int
        while ((len = inputStream.read(buf)) > 0) {
            outputStream.write(buf, 0, len)
        }
    }

    @Throws(IOException::class)
    fun copy(srcPath: String, destPath: String) {
        try {
            FileInputStream(srcPath).use { `is` -> FileOutputStream(destPath).use { os -> copy(`is`, os) } }
        } catch (e: IOException) {
            e.printStackTrace()
            throw e
        }

    }

    @Throws(IOException::class)
    fun copy(context: Context, srcUri: Uri, destPath: String) {
        try {
            context.contentResolver.openInputStream(srcUri)!!.use { `is` ->
                FileOutputStream(destPath).use { os ->
                    if (`is` != null) {
                        copy(`is`, os)
                    } else {
                        throw IOException("InputStream is null")
                    }
                }
            }
        } catch (e: IOException) {
            e.printStackTrace()
            throw e
        }

    }
}
